OpenTrons/opentrons-api
# feat(protocol-designer): off-deck labware visual representation

closes RAUT-696

# Overview

Now, there is a way to see visually which labware is off deck in PD at any given part of the protocol. And you can add off deck labware initially during the starting deck state.

# Test Plan

sandbox: http://sandbox.designer.opentrons.com/pd_off-deck-labware-in-setup/

- go to the setting and turn on the feature flag for enabling off deck visuals and multi tiprack support (keeping this on a new feature flag since this + multi tiprack are currently supported on the app and therefore, these features are very likely to be in PD 7.1).
- create a protocol for the Flex, go to the deck setup, click on the `edit off deck labware` button. The slideout should appear. Add a labware. Make sure the options to add nickname, add liquids, duplicate, delete are there and work.
- add a step to the protocol, click on the edit off deck labware button. Make sure the only option available is to view liquids. Make sure the button in the slideout is disabled.
- add a move labware step to move an off deck labware to the deck. Make sure the labware entity disappears from the slideout.
- repeat steps for OT-2 protocol
- export a protocol and in the json file, make sure the `loadlabware` that you loaded off deck initially says the location is off deck.

# Changelog

- make a slideout component in PD (its basically a copy of the app slideout)
- create `EditOffDeckLabware` 
- edit `browseLabware` and `NameThisLabware`'s prop to take in just the labware entity
- create new `OffDeckLabwareSlidoeut` and `OffDeckLabwareButton`
- some styling z-index changes

# Review requests

see test plan

# Risk assessment

low
